lib/page.js

```diff
@@ -133,10 +134,31 @@ class Page {
     this.title = await renderContent(this.rawTitle, context, { textOnly: true, encodeEntities: true })
     this.shortTitle = await renderContent(this.shortTitle, context, { textOnly: true, encodeEntities: true })
 
-    const markdown = this.mapTopic
+    let markdown = this.mapTopic
       ? getMapTopicContent(this, context.pages, context.redirects)
       : this.markdown
 
+    // If the article is interactive parse the React!
+    if (this.interactive) {
+      // Search for the react code comments to find the react components
+      const reactComponents = markdown.match(/<!--react-->(.*?)<!--end-react-->/gs)
+
+      // Render each of the react components in the markdown
+      await Promise.all(reactComponents.map(async (reactComponent) => {
+        let componentStr = reactComponent
+
+        // Remove the React comment indicators
+        componentStr = componentStr.replace('<!--react-->\n', '').replace('<!--react-->', '')
```

Comment:
Possible to use a RegExp with replace, like 
```suggestion
        componentStr = componentStr.replace(/<!--react-->\n?/g, '')
```

Reply:
Can you confirm this works @heiskr and if it does can you commit it? I get confused by the first replace having a new line character and the next one not. I personally like coding verbose. But that's a personal preference.

Reply:
I think its fine as is, just an FYI more than anything.